lib/components/viewers/live-stop-times.tsx

```diff
@@ -144,36 +169,72 @@ class LiveStopTimes extends Component<Props, State> {
       return stopTimeComparator(stopTimesA[0], stopTimesB[0])
     }
 
+    const routeTimes = Object.values(stopTimesByPattern)
+      .filter(({ times }) => times.length !== 0)
+      .sort(patternComparator)
+      .map((route) => {
+        const sortedTimes = route.times
+          .concat()
+          ?.sort(stopTimeComparator)
+          // Only show times within 24 hours of next arrival time
+          .filter((time: any, i: number, array: Array<any>) => {
+            const firstDepartureTime =
+              array[0].serviceDay + array[0].realtimeDeparture
+            const departureTime = time.serviceDay + time.realtimeDeparture
+            return i === 0 || (departureTime - firstDepartureTime) / 3600 < 24
+          })
+        const { serviceDay } = sortedTimes[0]
+        return {
+          ...route,
+          day: serviceDay,
+          times: sortedTimes
+        }
+      })
+      .filter(({ pattern, route }) =>
+        routeIsValid(route, getRouteIdForPattern(pattern))
+      )
+      .filter((route) => {
+        /* If the route's first departure time falls on the 2nd available service day,
+      only show it if it's within 6 hours of now. */
+        if (!isSameDay(utcToZonedTime(route.day * 1000, homeTimezone), now)) {
+          const departureTime = route.times[0].realtimeDeparture + route.day
+          return isBefore(departureTime * 1000, addHours(now, 6))
+        }
+        return route
```

Comment:
And ideally combine this one as well. There's some shared logic here as well so I think we can get clever with it